src/transformers/tokenization_utils_fast.py

```diff
@@ -191,27 +184,7 @@ def __init__(self, *args, **kwargs):
             token for token in self.all_special_tokens_extended if token not in encoder and token not in tokens_to_add
         ]
 
-        if len(tokens_to_add) > 0:
-            # super hack: if a token.special is set, tokenizer ignores it for now so FIXME @ArthurZ
-            # Accumulate added tokens into batches of special/non-special tokens, because calling add_tokens() for
-            # individual tokens would repeatedly rebuild a trie, which can be slow.
-            is_last_special = None
-            tokens = []
-            special_tokens = self.all_special_tokens
```

Comment:
this used to ensure that tokens get added as special if they are in `all_special_tokens`. It even overwrote the default `special=False` param when creating an `AddedToken` such as the mask token in fast models (like here: https://github.com/huggingface/transformers/blob/2b789f27f383435b8db2fee3d10b0a1358c0c234/src/transformers/models/layoutxlm/tokenization_layoutxlm_fast.py#L233-L234 

But I'm not sure why only the mask token gets added like this in so many models and if we should just pass `special=True` there in each model?

Reply:
yeah we should make them special

Reply:
Actually you are right, I kep the logic to check special / change special if in all special tokens

Reply:
All tests pass thanks to you